web-crawler/src/main/java/com/google/sps/webcrawler/WebCrawler.java

```diff
@@ -157,17 +157,17 @@ public String handleResponse(final HttpResponse response) throws IOException {
    */
   public Optional<NewsArticle> scrapeAndExtractHtml(String url) {
     try {
-      URL formattedUrl = new URL(url);
-      String robotsUrl = String.format("%s://%s/robots.txt", formattedUrl.getProtocol(),
-          formattedUrl.getHost());
-      InputStream robotsTxtStream = new URL(robotsUrl).openStream();
+      URL formattedUrlObj = new URL(url);
+      URL robotsUrlObj = new URL(formattedUrlObj.getProtocol(), formattedUrlObj.getHost(),
+          "/robots.txt");
```

Comment:
Also, I would double check whether the leading `/` is necessary?

Reply:
Yes, otherwise `robotsUrlObj.toString()` doesn't return the correct URL string.

Reply:
Cool, thanks for confirming!